src/porepy/viz/data_saving_model_mixin.py

```diff
@@ -315,3 +289,190 @@ def load_data_from_pvd(
         self.time_manager.load_time_information(times_file)
         self.time_manager.set_time_and_dt_from_exported_steps(time_index)
         self.exporter._time_step_counter = time_index
+
+
+class IterationExporting(pp.PorePyModel):
+    if TYPE_CHECKING:
+        data_to_export: Callable[[], Any]
+
+    def initialize_data_saving(self):
+        """Initialize iteration exporter."""
+        super().initialize_data_saving()
+        # Having a separate exporter for iterations avoids distinguishing between
+        # iterations and time steps in the regular exporter's history.
+        self.iteration_exporter = pp.Exporter(
+            self.mdg,
+            file_name=self.params["file_name"],
+            folder_name=self.params["folder_name"] + "_iterations",
+        )
+
+    def data_to_export_iteration(self):
+        """Returns data for iteration exporting.
+
+        Override to customize data to be exported at each nonlinear iteration.
+
+        Returns:
+            Any type compatible with data argument of pp.Exporter().write_vtu().
+
+        """
+        return self.data_to_export()
+
+    def save_data_iteration(self):
+        """Export current solution to vtu files.
+
+        This method is typically called by after_nonlinear_iteration.
+        """
+        # To make sure the nonlinear iteration index does not interfere with the time
+        # part, we multiply the latter by the next power of ten above the maximum number
+        # of nonlinear iterations. Default value set to 10 in accordance with the
+        # default value used in NewtonSolver.
+        n = self.params.get("max_iterations", 10)
```

Comment:
This type of doubly hard-coded default values is bound to fail at some point, though the consequences here should not be severe. I'm fine with leaving it for now, but it would be better to have a single location for all such default values. TODO, I guess?